lib/Service/OpenAiSettingsService.php

```diff
@@ -99,78 +100,84 @@ public function getUserApiKey(?string $userId, bool $fallBackOnAdminValue = fals
 	}
 
 	/**
 	 * @return string
 	 */
 	public function getAdminDefaultImageSize(): string {
-		return $this->config->getAppValue(Application::APP_ID, 'default_image_size') ?: Application::DEFAULT_DEFAULT_IMAGE_SIZE;
+		return $this->appConfig->getValueString(Application::APP_ID, 'default_image_size') ?: Application::DEFAULT_DEFAULT_IMAGE_SIZE;
 	}
 
 	/**
 	 * @return string
 	 */
 	public function getServiceUrl(): string {
-		return $this->config->getAppValue(Application::APP_ID, 'url');
+		return $this->appConfig->getValueString(Application::APP_ID, 'url');
 	}
 
 	/**
 	 * @return string
 	 */
 	public function getServiceName(): string {
-		return $this->config->getAppValue(Application::APP_ID, 'service_name');
+		return $this->appConfig->getValueString(Application::APP_ID, 'service_name');
 	}
 
 	/**
 	 * @return int
 	 */
 	public function getRequestTimeout(): int {
-		return intval($this->config->getAppValue(Application::APP_ID, 'request_timeout', strval(Application::OPENAI_DEFAULT_REQUEST_TIMEOUT))) ?: Application::OPENAI_DEFAULT_REQUEST_TIMEOUT;
+		return intval($this->appConfig->getValueString(Application::APP_ID, 'request_timeout', strval(Application::OPENAI_DEFAULT_REQUEST_TIMEOUT))) ?: Application::OPENAI_DEFAULT_REQUEST_TIMEOUT;
 	}
 
 	/**
 	 * @return int
 	 */
 	public function getMaxTokens(): int {
-		return intval($this->config->getAppValue(Application::APP_ID, 'max_tokens', strval(Application::DEFAULT_MAX_NUM_OF_TOKENS))) ?: Application::DEFAULT_MAX_NUM_OF_TOKENS;
+		return intval($this->appConfig->getValueString(Application::APP_ID, 'max_tokens', strval(Application::DEFAULT_MAX_NUM_OF_TOKENS))) ?: Application::DEFAULT_MAX_NUM_OF_TOKENS;
 	}
 
 	/**
 	 * @return string
 	 */
 	public function getLlmExtraParams(): string {
-		return $this->config->getAppValue(Application::APP_ID, 'llm_extra_params');
+		return $this->appConfig->getValueString(Application::APP_ID, 'llm_extra_params');
 	}
 
 	/**
 	 * @return int
 	 */
 	public function getQuotaPeriod(): int {
-		return intval($this->config->getAppValue(Application::APP_ID, 'quota_period', strval(Application::DEFAULT_QUOTA_PERIOD))) ?: Application::DEFAULT_QUOTA_PERIOD;
+		return intval($this->appConfig->getValueString(Application::APP_ID, 'quota_period', strval(Application::DEFAULT_QUOTA_PERIOD))) ?: Application::DEFAULT_QUOTA_PERIOD;
 	}
 
 	/**
 	 * @return int[]
 	 */
 	public function getQuotas(): array {
-		$quotas = json_decode($this->config->getAppValue(Application::APP_ID, 'quotas', json_encode(Application::DEFAULT_QUOTAS)) ?: json_encode(Application::DEFAULT_QUOTAS), true);
+		$quotas = json_decode(
+			$this->appConfig->getValueString(
+				Application::APP_ID, 'quotas',
+				json_encode(Application::DEFAULT_QUOTAS)
+			) ?: json_encode(Application::DEFAULT_QUOTAS),
+			true,
+		);
 		// Make sure all quota types are set in the json encoded app value (in case new quota types are added in the future)
 		if (count($quotas) !== count(Application::DEFAULT_QUOTAS)) {
```

Comment:
null check missing on json_decode's output

Reply:
There shouldn't be any case where the stored value is not valid JSON but let's be careful, I agree.